src/main/java/com/nutomic/syncthingandroid/util/Logger.java

```diff
@@ -0,0 +1,54 @@
+package com.nutomic.syncthingandroid.util;
+
+import android.widget.TextView;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Iterator;
+import java.util.LinkedList;
+
+public class Logger {
+
+    private final static String TAG = "Logger";
```

Comment:
I'd put this in `SyncthingService`, pass it to `SyncthingRunnable` in constructor, and provide a getter method (probably just returning the entire log string).

`getAndroidLog()` can go into the Activity imo.